utils/outputwriter/outputcontent.go

```diff
@@ -111,295 +150,705 @@ func untitledForJasMsg(writer OutputWriter) string {
-		ids = append(ids, cve.Id)
+func getDecoratorWithLicenseViolationTitle(writer OutputWriter) CommentDecorator {
+	return func(commentCount int, content string) string {
+		contentBuilder := strings.Builder{}
+		// Decorate each part of the split content with a title as prefix and return the content
+		WriteContent(&contentBuilder, writer.MarkAsTitle(licenseViolationTitle, 3))
+		WriteContent(&contentBuilder, content)
+		return contentBuilder.String()
 	}
-	return
 }
 
-type vulnerabilityOrViolationDetails struct {
-	details           string
-	title             string
-	dependencyName    string
-	dependencyVersion string
+func getLicenseViolationsSummaryTable(licenses []formats.LicenseViolationRow, writer OutputWriter) string {
+	table := NewMarkdownTable("Severity", "License", "Direct Dependencies", "Impacted Dependency", "Watch Name").SetDelimiter(writer.Separator())
+	if _, ok := writer.(*SimplifiedOutput); ok {
+		// The values in this cell can be potentially large, since SimplifiedOutput does not support tags, we need to show each value in a separate row.
+		// It means that the first row will show the full details, and the following rows will show only the direct dependency.
+		// It makes it easier to read the table and less crowded with text in a single cell that could be potentially large.
+		table.GetColumnInfo("Direct Dependencies").ColumnType = MultiRowColumn
+	}
+	for _, license := range licenses {
+		table.AddRowWithCellData(
+			NewCellData(writer.FormattedSeverity(license.Severity, "Applicable")),
+			NewCellData(license.LicenseKey),
+			getDirectDependenciesCellData(license.Components),
+			NewCellData(results.GetDependencyId(license.ImpactedDependencyName, license.ImpactedDependencyVersion)),
+			NewCellData(license.Watch),
+		)
+	}
+	return writer.MarkInCenter(table.Build())
 }
 
-func vulnerabilityDetailsContent(vulnerabilities []formats.VulnerabilityOrViolationRow, writer OutputWriter) (content []string) {
-	vulnerabilitiesWithDetails := getVulnerabilityWithDetails(vulnerabilities)
-	if len(vulnerabilitiesWithDetails) == 0 {
+func getLicenseViolationsDetailsContent(licenseViolations []formats.LicenseViolationRow, writer OutputWriter) (content []string) {
+	if len(licenseViolations) == 0 {
 		return
 	}
-	// Prepare content for each vulnerability details
-	for i := range vulnerabilitiesWithDetails {
-		if len(vulnerabilitiesWithDetails) == 1 {
-			content = append(content, vulnerabilitiesWithDetails[i].details)
+	for _, violation := range licenseViolations {
+		if len(licenseViolations) == 1 {
+			content = append(content, getScaLicenseViolationDetails(violation, writer))
 		} else {
-			content = append(content, writer.MarkAsDetails(
-				fmt.Sprintf(`%s %s %s`, vulnerabilitiesWithDetails[i].title,
-					vulnerabilitiesWithDetails[i].dependencyName,
-					vulnerabilitiesWithDetails[i].dependencyVersion),
-				4, vulnerabilitiesWithDetails[i].details,
+			content = append(content, "\n"+writer.MarkAsDetails(
```

Comment:
please extract the inner calls to variables and only then place them in the func. It is extreamly hard to read what is going on here. Id rather it to be longer but so we can understand it next time we come back to it